### PR TITLE
[relationships] add edge mount  relationship between Hashicorp vault and k8's secrets 

### PR DESCRIPTION
**Pull request by deepak4566**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
🥇

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+     "allow": {
+        "from": [
+          {
+            "kind": "Secret",
+            "model": "kubernetes",
+              "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                [
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:32]:
```suggestion
                ["name"],
                ["namespace"]
```

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        "to": [
+          {
+            "kind": "VaultBinding",
+            "model": "k8svault-controller",
+             "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                [
+                  "spec.secret.name",
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:49]: This is not going to work. 
Update the path to `["settings" ,"spec", "secret", "name"]`.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                  "spec.secret.namespace"
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:50]: Same.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          }
+        ],
+
+        "to": [
+          {
+            "kind": "VaultBinding",
+            "model": "k8svault-controller",
+             "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:47]: This field is for the Inventory relationship.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
+{
+  "apiVersion": "vault.infra.doodle.com/v1beta1",
+  "kind": "Edge",
+  "metadata": {
+    "description": ""
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:5]: Please add description.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  "model": {
+    "name": "k8svault-controller",
+    "version": "0.1.2",
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:8]: Will this relationship work for `version: "0.1.2"`, but not for `version: "1.1.155"` ?

Comment (RipulHandoo) [reply server/meshmodel/k8svault-controller/relationships/edge_relationship.json:8]: If it works for both use `version: "*"`.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    "displayName": "K8s Vault Controller",
+    "category": {
+      "name": "Security \u0026 Compliance",
+      "metadata": null
+    },
+    "metadata": {}
+  },
+   "subType": "Mount",
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:16]: Could you please ensure that you are specifying the relationship for subType -> Mount? Currently, I don't see any components being mounted.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+              ],
+           }
+          }
+        ],
+
+        "to": [
+          {
+            "kind": "VaultBinding",
+            "model": "k8svault-controller",
+             "patch": {
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:45]: These fields won't work for `Mount` relationship.

File: server/meshmodel/k8svault-controller/relationships/edge_relationship.json [hunk 9]
```diff
@@ -0,0 +1,61 @@
+{
```
Comment (RipulHandoo) [inline server/meshmodel/k8svault-controller/relationships/edge_relationship.json:1]: Please rename the file. It should be `edge_<relationship_subType>.json`

**Review by RipulHandoo — CHANGES_REQUESTED** [review]

**Comment by RipulHandoo** [comment]
@deepak4566, could you please provide a working proof of this PR? A screen recording would be preferred.

**Comment by RipulHandoo** [comment]
If you're considering setting up a Mount relationship, I found this resource helpful: https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar.
 It offers guidance on injecting Vault secrets into Kubernetes pods via a sidecar. Just a heads-up, this is just one reference among many available. Feel free to explore other documentation and resources as well.

**Review by leecalcote — CHANGES_REQUESTED** [review]
Is this supposed to have a third component implicated? Which of the three components is to sit atop the edge?

Will you contrast this against the existing PVC mount relationship?

A description of the relationship is still missing.

**Comment by leecalcote** [comment]
Have you tested this relationship? Does it mutate the implicated components as expected? Are you able to deploy Vault and bind the Secret successfully with this relationship? 

@deepak4566, please demo this at tomorrow's Meshery Dev meeting. Add the agenda, if you would.

**Comment by deepak4566** [comment]
> Have you tested this relationship? Does it mutate the implicated components as expected? Are you able to deploy Vault and bind the Secret successfully with this relationship?
> 
> @deepak4566, please demo this at tomorrow's Meshery Dev meeting. Add the agenda, if you would.

i think at present this relationship doesnt work , before this relationship between secrets i think there is need of some other relationships like mount service account jwt token to vault and role , and many more .

**Pull request closed without merging**